src/js/components/CollapsibleErrorMessage.js

```diff
@@ -0,0 +1,110 @@
+
+import Icon from './Icon';
+
+import React from 'react';
+import classNames from 'classnames';
+
+/**
+ * An error bar component commonly used in the ServiceForm and the JobsForm
+ * that contains collapsible error messages.
+ */
+class CollapsibleErrorMessage extends React.Component {
+
+  /**
+   * Initialize superclass
+   */
+  constructor() {
+    super(...arguments);
+
+    // Initial state
+    this.state = {
+      expanded: false
+    };
+
+  }
+
+  /**
+   * Toggle the expanded state of the collapsible error bar
+   */
+  toggleExpanded() {
+    this.setState({
+      expanded: !this.state.expanded
+    });
+  }
+
+  /**
+   * React.js Render Function
+   *
+   * @returns {React.Component} - The rendered content
+   */
+  render() {
+    let {message, details} = this.props;
+    let isVisible = !!message;
+    let isDetailed = isVisible && details && details.length;
+    let contents = [];
+
+    // For sure, create a message body
+    contents.push(
+        <div key="fixed" className="collapsible-fixed text-align-center text-danger flush-top">
+          <Icon
+            family="mini"
+            id="yield"
+            size="mini"
+            className="icon-alert icon-margin-right"
+            color="red" />
+          {message}
+          {(() => {
```

Comment:
Here for an example for how we separate logic from view definition: https://github.com/dcos/dcos-ui/blob/master/src/js/components/ImageViewer.js#L45

```js
const METHODS_TO_BIND = [
  'toggleExpanded'
];

...
  constructor() {
    super(...arguments);

    this.state = {
      expanded: false
    };

    // We use our bind snippet, when context needs to be changed
    METHODS_TO_BIND.forEach((method) => {
      this[method] = this[method].bind(this);
    });
  }

  ...

  getShowDetailLink() {
    // Base cases return early
    if (!isDetailed) {
      return null;
    }
    let text = 'more';
    // We don't use ternary operators
    if (this.state.expanded) {
      text = 'less';
    }

    return (
       <span>
         &nbsp;
         <span
           className="text-danger clickable"
           onClick={this.toggleExpanded}>
           <small>{`(Show ${text})`}</small>
         </span>
       </span>
    );
  }
```